src/ctia/lib/riemann.clj

```diff
@@ -0,0 +1,124 @@
+(ns ctia.lib.riemann
+  (:require [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [riemann.client :as riemann]
+            [ctia.properties :as prop]
+            [ctia.lib.utils :as utils])
+  (:import [clojure.lang ExceptionInfo]))
+
+(defn request->event
+  [request extra-fields]
+  (into {:uri (str (:uri request))
+         :_params (utils/safe-pprint (:params request))
+         :remote-addr (str (if-let [xff (get-in request [:headers "x-forwarded-for"])]
+                             (peek (str/split xff #"\s*,\s*"))
+                             (:remote-addr request)))
+         :request-headers (prn-str (:headers request))
```

Comment:
This look dangerous. You should take care to get rid of the `authorization` HEADER before printing it.
There is a good reason not to show the headers in IROH.

In fact I would rather prefer having flattened headers and taking care of removing sensitive headers.
Also there should be only selected headers that should be flattened for all other headers this should be pretty printed.

So typically:

- `:request-header-content-type` ".."
- `:request-header-authorization-type "Bearer"` (hide the JWT/API key but keep the bearer name if it exists, so that could be "Bearer" "api-token" "api-key" or nothing if the header value does not contain any space. So the Authorization header contain a value like "Bearer <JWT>" or "api-token <api-token>" or "<key>", I think it is useful to keep track of the authorization type that could be "bearer", "api-token", etc... or "apikey")
- `:request-header-origin`
- `:request-header-referrer`
- ...

And for all other headers: `:request-headers-unkown (pr-str (dissoc (:headers req) known-headers))`.
If you don't do that a client could send CURL request with random string for header value and completely fill the index with random keys. And we should really take care about the keys of the events we send cannot grow too much.

Reply:
I deleted the extra stuff, now just has a subset of the functionality of riemann-reporter. Could you take another look to let me know if your concern is still relevant, it's too much information for me to process.